projects/observability/src/shared/dashboard/data/graphql/table/entity/entity-table-data-source.model.ts

```diff
@@ -117,15 +105,26 @@ export class EntityTableDataSourceModel extends TableDataSourceModel {
 
   private queryEntityChildren(
     entity: Entity,
-    request: TableDataRequest<SpecificationBackedTableColumnDef>
+    parentRequest: TableDataRequest<SpecificationBackedTableColumnDef>
   ): Observable<TableRow[]> {
-    return this.query<EntitiesGraphQlQueryHandlerService>(this.buildChildEntityRequest(request, entity)).pipe(
-      map(response => this.resultsAsTreeRows(response.results, request, false))
+    this.childEntityDataSource!.parentEntityFilter = GraphQlEntityFilter.forEntity(entity);
```

Comment:
Do we need to introduce state for this? We could just as easily introduce a private method like
`private getDataForParentEntity(entity: Entity): Observable<TableRow[]>`

Reply:
What would that do? I need a way for the parent to apply an entity filter on parent id.  This filter would change as user clicks on different rows.

Reply:
So I'm suggesting instead of
```typescript
this.childEntityDataSource!.parentEntityFilter = GraphQlEntityFilter.forEntity(entity);
return this.childEntityDataSource!.getData()...
```
we update it to
```typescript
return this.childEntityDataSource!.getDataForParentEntity(entity)...
```

It would work pretty much the same, it just prevents any bugs coming from missing assumptions between the caller and callee side since now those are represented in the function contract.

Reply:
If i can convert this entity filter into a table filter, then the code will be much cleaner. I can just add it to the request generated below.

Reply:
That works too - It's the same code either way right? Just whether we convert the entity to a filter in the parent v child?
I was favoring passing an entity instead of passing a filter because it felt like it kept the responsibility for how to build a scoped request all in the child data source rather than split between the parent and child.

Reply:
Cool. Makes sense. 

Can we also modify the `getData` of the Entity-table-data-source or Table data source itself, to take an optional entity filter? Something like this

```
  public getData(entityFilter?: GraphqlFilter): Observable<TableDataSource<TableRow, SpecificationBackedTableColumnDef>> {
    return observableOf({
      getData: request =>
        this.query(filters => this.buildGraphQlRequest(filters.concat(entityFilter), request)).pipe(
          map(response => this.buildTableResponse(response, request))
        ),
      getScope: () => this.getScope()
    });
  }
```

Reply:
We would still have to build the child request in parent though. Example, this new request should have the correct page start index and limit and the column specifications.

Reply:
I thought about overloading getData, but it felt potentially confusing since that api is public and called externally, while keeping a separate one allows us to keep it private and the argument required.

Reply:
i think the code will be much cleaner if I can convert the parent entity filter into a table filter and pass it in the child table request.

Reply:
But there is no easy way to do it (I think)

Reply:
done